ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -29,7 +29,44 @@
 from rosidl_runtime_py import get_service_interfaces
 
 
-def generate_interfaces(interfaces, html_dir, template, interface_type):
+def generate_interfaces_index(messages, services, actions, html_dir, timestamp):
+    """
+    Generate index for packages.
+
+    :param messages: Dictionay with the message package as a key and a list message names as value
+    :param type: dict
+    :param services: Dictionay with the service package as a key and a list sercice names as value
+    :param type: dict
+    :param actions: Dictionay with the action package as a key and a list action names as value
+    :param type: dict
+    :param html_dir: path to the directory to save the index
+    :type html_dir: str
+    :param timestamp: timestamp to include in the index site
+    :param timestamp: time.struct_time
+    """
+    interface_packages = list(messages.keys()) + list(services.keys()) + list(actions.keys())
+
+    for package_name in interface_packages:
+        msg_list = []
+        srv_list = []
+        action_list = []
+        if package_name in messages.keys():
+            msg_list = messages[package_name]
+        if package_name in services.keys():
+            srv_list = services[package_name]
+        if package_name in actions.keys():
+            action_list = actions[package_name]
+        interface_packages_dict = {
+            'msg': msg_list,
+            'srv': srv_list,
+            'action': action_list,
+        }
+        package_directory = os.path.join(html_dir, package_name)
+        os.makedirs(package_directory, exist_ok=True)
+        utils.generate_index(package_name, package_directory, interface_packages_dict, timestamp)
```

Comment:
What do you think about passing msg_list, srv_list and action_list as optional named parameters into generate_index?